temporalio/worker/workflow_sandbox/restrictions.py

```diff
@@ -0,0 +1,971 @@
+    "argparse,array,ast,asynchat,asyncio,asyncore,atexit,audioop,base64,bdb,binascii,binhex,"
+    "bisect,builtins,bz2,cProfile,calendar,cgi,cgitb,chunk,cmath,cmd,code,codecs,codeop,"
+    "collections,colorsys,compileall,concurrent,configparser,contextlib,contextvars,copy,"
+    "copyreg,crypt,csv,ctypes,curses,dataclasses,datetime,dbm,decimal,difflib,dis,distutils,"
+    "doctest,email,encodings,ensurepip,enum,errno,faulthandler,fcntl,filecmp,fileinput,"
+    "fnmatch,fractions,ftplib,functools,gc,genericpath,getopt,getpass,gettext,glob,graphlib,"
+    "grp,gzip,hashlib,heapq,hmac,html,http,idlelib,imaplib,imghdr,imp,importlib,inspect,"
+    "io,ipaddress,itertools,json,keyword,lib2to3,linecache,locale,logging,lzma,mailbox,"
+    "mailcap,marshal,math,mimetypes,mmap,modulefinder,msilib,msvcrt,multiprocessing,netrc,"
+    "nis,nntplib,nt,ntpath,nturl2path,numbers,opcode,operator,optparse,os,ossaudiodev,"
+    "pathlib,pdb,pickle,pickletools,pipes,pkgutil,platform,plistlib,poplib,posix,posixpath,"
+    "pprint,profile,pstats,pty,pwd,py_compile,pyclbr,pydoc,pydoc_data,pyexpat,queue,quopri,"
+    "random,re,readline,reprlib,resource,rlcompleter,runpy,sched,secrets,select,selectors,"
+    "shelve,shlex,shutil,signal,site,smtpd,smtplib,sndhdr,socket,socketserver,spwd,sqlite3,"
+    "sre_compile,sre_constants,sre_parse,ssl,stat,statistics,string,stringprep,struct,"
+    "subprocess,sunau,symtable,sys,sysconfig,syslog,tabnanny,tarfile,telnetlib,tempfile,"
+    "termios,textwrap,this,threading,time,timeit,tkinter,token,tokenize,trace,traceback,"
+    "tracemalloc,tty,turtle,turtledemo,types,typing,unicodedata,unittest,urllib,uu,uuid,"
+    "venv,warnings,wave,weakref,webbrowser,winreg,winsound,wsgiref,xdrlib,xml,xmlrpc,zipapp,"
+    "zipfile,zipimport,zlib,zoneinfo"
+)
+
+SandboxRestrictions.passthrough_modules_maximum = SandboxRestrictions.passthrough_modules_with_temporal | SandboxMatcher(
+    access={
+        # All stdlib modules except "sys" and their children. Children are not
+        # listed in stdlib names but we need them because in some cases (e.g. os
+        # manually setting sys.modules["os.path"]) they have certain child
+        # expectations.
+        v
+        for v in _stdlib_module_names.split(",")
+        if v != "sys"
+    }
+)
+
+SandboxRestrictions.passthrough_modules_default = (
+    SandboxRestrictions.passthrough_modules_maximum
+)
+
+
+def _public_callables(parent: Any, *, exclude: Set[str] = set()) -> Set[str]:
+    ret: Set[str] = set()
+    for name, member in inspect.getmembers(parent):
+        # Name must be public and callable and not in exclude and not a class
+        if (
+            not name.startswith("_")
+            and name not in exclude
+            and callable(member)
+            and not inspect.isclass(member)
+        ):
+            ret.add(name)
+    return ret
+
+
+# TODO(cretz): Should I make this more declarative in an external file?
```

Comment:
I think this is fine honestly, especially if you could change it slightly to allow passing just a string list for all the `all_uses` cases, which would reduce the verbosity a lot.

Reply:
Yeah, I may make it a bit easier to write. I am not happy with my `SandboxMatcher` and I have learned a lot since, but I marked this entire API as unstable, so I can clean later.